apps/full-site-editing/full-site-editing-plugin/block-inserter-modifications/contextual-tips/contextual-tip.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { get, filter, deburr, lowerCase, includes, uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Tip } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import tipsList from './list';
+
+export default function ( { searchTerm, random = false } ) {
+	if ( ! searchTerm ) {
+		return null;
+	}
+
+	if ( ! tipsList.length ) {
+		return null;
+	}
+
+	const normalizedSearchTerm = deburr( lowerCase( searchTerm ) ).replace( /^\//, '' );
+
+	const foundTips = filter(
+		tipsList,
+		( { keywords, permission } ) =>
+			permission() &&
+			filter( uniq( keywords ), ( keyword ) => includes( normalizedSearchTerm, keyword ) ).length
+	);
+
+	if ( ! foundTips.length ) {
+		return null;
+	}
+
+	const index = random ? Math.floor( Math.random() * foundTips.length ) : 0;
```

Comment:
When would we want to show a random tip?

Reply:
when there are more that one coincidence in the search term:

<img src="https://user-images.githubusercontent.com/77539/81800053-9cab0a80-94e8-11ea-9574-ecafc073eaa5.png" width="300px" />